### PR TITLE
Git tags - IO1-2235

### PR DESCRIPTION
**Pull request by mtabasco**
Add  git tag support.
- On every deployment and upgrade, the the tagging process is presented
- Tag naming follows semantic versioning

TODO: After merging [Versioned contract](https://github.com/bloxapp/ssv-network/pull/179), the tag created will be passed as the version of the contract to the `initialize` function so tags and contract versions are in sync.

### CODE REVIEW

**Review by mtabasco — COMMENTED** [review]
all done

File: scripts/validate-upgrade-ssv-network-views.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
+import { ethers, upgrades } from 'hardhat';
+
 async function validateUpgradeSSVNetworkViews() {
-  const proxyAddress = process.env.SSVNETWORKVIEWS_PROXY_ADDRESS;
+  const proxyAddress: any = process.env.SSVNETWORKVIEWS_PROXY_ADDRESS;
```
Comment (andrew-blox) [inline scripts/validate-upgrade-ssv-network-views.ts:4]: no any :)

File: scripts/validate-upgrade-ssv-network.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
+import { ethers, upgrades } from 'hardhat';
+
 async function validateUpgradeSSVNetwork() {
-  const proxyAddress = process.env.SSVNETWORK_PROXY_ADDRESS;
-  const SSVNetwork = await ethers.getContractFactory("SSVNetwork_V2");
+  const proxyAddress: any = process.env.SSVNETWORK_PROXY_ADDRESS;
```
Comment (andrew-blox) [inline scripts/validate-upgrade-ssv-network.ts:4]: again

**Review by andrew-blox — CHANGES_REQUESTED** [review]

**Comment by andrew-blox** [comment]
I have not tested this live yet just reviewed the code.

File: scripts/validate-upgrade-ssv-network.ts (hunk 2 above)
Comment (mtabasco) [reply scripts/validate-upgrade-ssv-network.ts:4]: solved in `2-step-transfer` PR, thanks!

**Comment by lior-blox** [comment]
@mtabasco few things:
1. use the version in the initializer in the contract
2. when selecting an existing/creating tag, git checkout from the tag and deploy that code

File: README.md [hunk 3]
```diff
@@ -1,41 +1,29 @@
...
   
 
 This repository contains a SSVNetwork smart contacts project.
 
-  
-
 ## Quick start
-
-  
+
```
Comment (lior-blox) [inline README.md:7]: .

**Review by lior-blox — CHANGES_REQUESTED** [review]
.

**Pull request closed without merging**